src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECDiffieHellmanCng.cs

```diff
@@ -43,10 +40,7 @@ public CngAlgorithm HashAlgorithm
 
             set
             {
-                if (_hashAlgorithm == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
+                ArgumentNullException.ThrowIfNull(value, nameof(value));
 
```

Comment:
Seems like maybe `set!!` should be part of the new syntax?

Reply:
@RikkiGibson and I were chatting about that earlier this week.  I think he was going to bring it up in LDM.